src/scaffoldfitter/fitterstepalign.py

```diff
@@ -279,6 +279,31 @@ def _doAutoAlign(self):
 
         self._optimiseAlignment(pointMap)
 
+    def getTransformationMatrix(self):
+        '''
+        :return: 4x4 row-major transformation matrix with first index down rows, second across columns,
+        suitable for multiplication p' = Mp where p = [ x, y, z, h ].
+        '''
+        # apply transformation in order: scale then rotation then translation
+        if not all((v == 0.0) for v in self._rotation):
+            rotationMatrix = euler_to_rotation_matrix(self._rotation)
+            return [
+                [rotationMatrix[0][0]*self._scale, rotationMatrix[0][1]*self._scale, rotationMatrix[0][2]*self._scale, self._translation[0]],
+                [rotationMatrix[1][0]*self._scale, rotationMatrix[1][1]*self._scale, rotationMatrix[1][2]*self._scale, self._translation[1]],
+                [rotationMatrix[2][0]*self._scale, rotationMatrix[2][1]*self._scale, rotationMatrix[2][2]*self._scale, self._translation[2]],
+                [0.0, 0.0, 0.0, 1.0]]
+        if not (self._scale == 1.0 and all((v == 0.0) for v in self._translation)):
+            return [
+                [self._scale, 0.0, 0.0, self._translation[0]],
+                [0.0, self._scale, 0.0, self._translation[1]],
+                [0.0, 0.0, self._scale, self._translation[2]],
+                [0.0, 0.0, 0.0, 1.0]]
+        return [
+            [1.0, 0.0, 0.0, 0.0],
+            [0.0, 1.0, 0.0, 0.0],
+            [0.0, 0.0, 1.0, 0.0],
+            [0.0, 0.0, 0.0, 1.0]]
+
```

Comment:
I have added a `matrix_identity` method to `vectorops`, can you use that instead of explicitly making it here.